sale_order_line_reorder_fields/models/sale_order_line.py

```diff
@@ -4,23 +4,41 @@ class SaleOrderLineExtend(models.Model):
     _inherit = 'sale.order.line'
 
     call_rate = fields.Float('Call Rate')
+    updating_discount  = fields.Boolean()
+    updating_call_rate  = fields.Boolean()
     
     @api.onchange('discount')
-    def change_discount(self):
+    def _onchange_discount(self):
         for line in self:
             if line.discount:
                 line.call_rate = 100 - line.discount
-                
+                line.updating_discount = True
+
     @api.onchange('call_rate')
-    def change_call_rate(self):
+    def _onchange_call_rate(self):
         for line in self:
             if line.call_rate:
                 line.discount = 100 - line.call_rate
-                
-    @api.onchange('product_id', 'price_unit', 'product_uom', 'product_uom_qty', 'tax_id')
-    def _onchange_discount(self):
-        super(SaleOrderLineExtend, self)._onchange_discount()
-
-        if self.discount is not None:
+                line.updating_call_rate = True
+    
+    @api.onchange('product_id')
+    def onchange_product_id(self):
+        self.ensure_one()
+        if self.updating_discount or self.updating_call_rate:
+            return
+        elif self.order_id and self.order_id.pricelist_id and self.order_id.pricelist_id.discount_policy == 'without_discount' and self.order_id.pricelist_id.item_ids:
+            last_item = self.order_id.pricelist_id.item_ids[-1]
+            price_str = last_item.price
+            percent_value = float(price_str.split()[0].replace('%', ''))
+            
+            self.discount = percent_value
             self.call_rate = 100 - self.discount
-            
\ No newline at end of file
+        
+    def write(self, vals):
+        res = super(SaleOrderLineExtend, self).write(vals)
+        for line in self:
+            if line.call_rate and line.discount:
+                return
```

Comment:
ở đây ko cần return res à?

Reply:
![image](https://github.com/ezt-togawa/rtw-custom/assets/127169715/779336a5-c7fe-4c63-9898-7f2c45417c61)
nó có rùi á anh
git nó ko hiển thị thì phải

Reply:
ko trong vòng for kia nếu cái if đầu tiên thỏa mãn thì đang return luôn.
trong t.hợp này thì có sao ko?

Reply:
nãy em check nó vào trong này cũng ko sao 
vậy e để break cho chắc ăn

Reply:
anh check lại hộ em nhé